src/gui/torrentcreatordialog.cpp

```diff
@@ -170,7 +172,7 @@ void TorrentCreatorDialog::dragEnterEvent(QDragEnterEvent *event)
 // Main function that create a .torrent file
 void TorrentCreatorDialog::onCreateButtonClicked()
 {
-    const auto inputPath = Utils::Fs::toCanonicalPath(Path(m_ui->textInputPath->text().trimmed()));
+    const auto inputPath = Path(m_ui->textInputPath->text().trimmed());
```

Comment:
I think this is the only change you need in this file. Please revert changes to QFileDialog.

I've briefly tested my suggestion and it works for me (on linux), please verify it on your system too.

Reply:
On macOS, I'm unable to select a symlink in the first place (its target gets selected instead) unless `QFileDialog::DontResolveSymlinks` is specified.

By the way, is `.trimmed()` really needed here? Paths ending with whitespace are perfectly valid, after all...

Reply:
>On macOS, I'm unable to select a symlink in the first place (its target gets selected instead) unless QFileDialog::DontResolveSymlinks is specified.

Even so, will it canonicalize your path? On my linux, it won't. It will still return the path with symlink in it. Note that this is tested without `QFileDialog::DontResolveSymlinks` flag.

>By the way, is .trimmed() really needed here? Paths ending with whitespace are perfectly valid, after all...

You are right and I have ideas to improve it. Lets leave it as is for now.

Reply:
>It's strange that it's not the case with Linux, but I guess the flag won't hurt anyways.

OK.

One last question. I just tested on Windows. Before this PR I can put a path ending with .lnk (which links to a folder) and successfully create a torrent. For example: `C:\Users\abc\Desktop\folder.lnk` Now it can't, it fails. The reason seems to be we don't canonicalize the path now.
So question is do we want to retain the previous behavior? Personally I think we should.

Reply:
My first guess would be to mimic the symlink behaviour, but if we keep the `.lnk` extension, it's gonna look strange, and otherwise, this path will probably be missing on the local FS. So, I guess, retaining the previous behaviour is the way to go.

Reply:
Actually I'm not sure about the "path without the extension will be missing on the local FS" part, if it's treated in a special way by the system APIs, then I think we should mimic the symlink behaviour.